subsys/bluetooth/host/keys.c

```diff
@@ -427,6 +427,19 @@ static int keys_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 		memcpy(keys->storage_start, val, len);
 	}
 
+	/* As of Core v6.2, authenticated keys are only valid for OOB or LE SC pairing
+	 * methods. This check ensures that keys are valid if a device is updated from a
+	 * previous version that did not enforce this requirement.
+	 */
+	if ((keys->flags & BT_KEYS_AUTHENTICATED) &&
+	    !((keys->flags & BT_KEYS_OOB) || (keys->flags & BT_KEYS_SC))) {
+		LOG_ERR("Authenticated keys for %s do not meet requirements",
+			bt_addr_le_str(&addr));
+		bt_keys_clear(keys);
+
```

Comment:
I think we should invoke `bt_conn_auth_info_cb.bond_deleted` to notify the application, in case the application needs to clean up something that's tied to the existence of the bond.

Reply:
Like this? https://github.com/zephyrproject-rtos/zephyr/blob/4a0e6d48c9726723a08ee2f6f497a7c53965da40/subsys/bluetooth/host/hci_core.c#L2179-L2186